third_party/move/move-vm/types/src/gas.rs

```diff
@@ -338,18 +338,44 @@ pub trait GasMeter: NativeGasMeter {
     // TODO(Gas): Expose the two elements
     fn charge_vec_swap(&mut self, ty: impl TypeView) -> PartialVMResult<()>;
 
+    /// Charge for fetching resource without the actual bytes
+    /// Used when loading the resource size without deserializing it
+    fn charge_resource_fetch(
+        &mut self,
+        addr: AccountAddress,
+        ty: &impl TypeView,
+        resource_exists: bool,
+        bytes_loaded: NumBytes,
+    ) -> PartialVMResult<()>;
+
+    /// Charge for deserialized bytes of a loaded resource
+    fn charge_loaded_bytes(
+        &mut self,
+        addr: AccountAddress,
+        ty: &impl TypeView,
+        val: impl ValueView,
+    ) -> PartialVMResult<()>;
+
     /// Charges for loading a resource from storage. This is only called when the resource is not
     /// cached.
     ///
     /// WARNING: This can be dangerous if you execute multiple user transactions in the same
     /// session -- identical transactions can have different gas costs. Use at your own risk.
+    ///
+    /// This function charges for both fetching the resource metadata and its bytes
     fn charge_load_resource(
         &mut self,
         addr: AccountAddress,
-        ty: impl TypeView,
+        ty: &impl TypeView,
         val: Option<impl ValueView>,
         bytes_loaded: NumBytes,
-    ) -> PartialVMResult<()>;
+    ) -> PartialVMResult<()> {
```

Comment:
Hmmmm, i think this has different behaviour from before? Note that we have in prod `MemoryTracker<StandardGasMeter<..>>` which means we were using `charge_loaded_bytes` first, then `charge_resource_fetch` if feature is off? I would suggest removing this method altogether, we can keep two, and keep the old ordering if feature is disabled, and new ordering (like here), if it is enabled